### DISCUSSION TITLE
Removing headers

### DISCUSSION THREAD

**Discussion by aurora-dot** [go-rod/rod#1065] [Q&A]
I know you can add headers, but is there any way to remove some?

Some scraper detectors look at not usually set headers and there are some which typical browsers don't have set, i would like to remove them.

Thank you!

**Comment by ysmood** [comment]
https://github.com/go-rod/rod/blob/0340433eab4e6551f075eaf1875035469a777480/hijack_test.go#L46

```go
r.Req().Header.Del("Header-You-Want-To-Delete")
```

**Reply by aurora-dot** [reply]
Thank you! I'll try to implement this soon!

**Reply by aurora-dot** [reply]
Made some progress, so the problem is where I set `accept-encoding`, does hijacking only allow certain encodings? 
Also, is there a way to order the headers? 
Thanks!


-- old message --
Hi! So I attempted to implement this however the response from calling `page.MustNavigate(event.Url).WaitStable(time.Duration(15))` after setting up the browser (using the function below) ends up with a strange response, removing the router or using the continue request function fixes this, however the request isn't altered if so. Do you know what I am doing wrong?

Thanks for the help so far <3

Function:
```go
func setupBrowser() (*rod.Page, *rod.HijackRouter) {
	CHROME_PATH := os.Getenv("CHROME_PATH")
	browserArgs := launcher.New()
	wsURL := browserArgs.Bin(CHROME_PATH).MustLaunch()
	browser := rod.New().ControlURL(wsURL).MustConnect().DefaultDevice(devices.IPhoneX)

	page := stealth.MustPage(browser)
	page.MustSetExtraHeaders(
		"Accept", "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8",
		"Accept-Language", "en-GB,en;q=0.5",
		"Accept-Encoding", "gzip, deflate, br, zstd",
	)

	router := page.HijackRequests()

	router.MustAdd("*", func(ctx *rod.Hijack) {
		r := ctx.Request
		r.Req().Header.Del("DEVICE-MEMORY")
		r.Req().Header.Del("DPR")
		r.Req().Header.Del("SEC-CH-PREFERS-COLOR-SCHEME")
		r.Req().Header.Del("SEC-CH-PREFERS-REDUCED-MOTION")
		ctx.MustLoadResponse()
	})

	go router.Run()

	return page, router
}
```
Response:
```
<html><head></head><body>��]�r�ƒ���3U��#�lيM�Qd��ƶK���֖jID ��$�Tab�Clm�fl��!׾��">���۔��f��/����R��[...]�RW7k��B���ׯ���9��w</jg*�rn��z�></o��%[></body></html>
```

**Reply by ysmood** [reply]
In golang `time.Duration(15)` means no time, you need to specify a unit for it.
